### PR TITLE
Fix hanging Hindcast tests

### PR DESCRIPTION
**Pull request by ssolson**
This PR addresses 2 issues with the hindcast tests:
1. On PRs tests were only being run on changes to hindcast test files.
    - Fixed to run on changes to hindcast test and module files.
2. The `prepare cache` job was failing due to 6 hour job limit.
    - Fixed by breaking out each hindcast cache test into its own job.

Additionally this PR updates action packages to the latest version where possible.

### CODE REVIEW

**Comment by ssolson** [comment]
Okay we are for sure hitting a 6 hour limit of the wind tests.

image

Running the tests locally it takes 4 hours and 22 minutes. Which is pretty insane amount of time. I think if I break these out into their own jobs I could get 6 hours per job, 

image

**Comment by ssolson** [comment]
Okay great the hindcast jobs now run separate which should fix the 6 hour issue. The new job flow looks as follows:
image

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -30,15 +30,15 @@ jobs:
...
 
       - name: Check for changes in wave/io/hindcast
         id: changes
-        uses: dorny/paths-filter@v2
+        uses: dorny/paths-filter@v3
         with:
           filters: |
             wave_io_hindcast:
-              - 'mhkit/tests/wave/io/hindcast/**'
+              - 'mhkit/wave/io/hindcast/**'
```
Comment (akeeste) [inline .github/workflows/main.yml:41]: we should probably check for changes in both places: hindcast tests and the hindcast module itself

Comment (ssolson) [reply .github/workflows/main.yml:41]: Thanks I have been too lazy to investigate if I could just add the test line to the filters.  Looks like I can.

I'll push the commit up after the current actions pass or fail.

**Comment by ssolson** [comment]
Okay great new test structure is working. Just need to check these final adjustments to make sure the check changes on multiple file locations and bumping up the versions on the actions works.

**Comment by ssolson** [comment]
@akeeste with the tests now working could you give this a final review/ approval (assuming all tests pass in ~4 hours)? 

My goal is to get this into develop today/ this weekend so I can then have these tests pass on #307 over the weekend.

**Pull request merged**